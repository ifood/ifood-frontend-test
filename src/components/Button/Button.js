import React from 'react'
import PropTypes from 'prop-types'

/* */

import styles from './Button.module.scss'

/* */

class Button extends React.Component {

    colorDefault(){

        return /\d/.test(this.props.color) || this.props.color === 'white' || this.props.color === 'black' ? this.props.color : this.props.color + '-500'

    }

    withLabel(){

        return (this.props.label !== '' && true) || ''

    }

    /* */

    render(){

        const colorDefault = this.colorDefault()
        const withLabel = this.withLabel()

        /* */

        return (

            <button

            onClick={ this.props.onClick }

            className={

                [

                    styles.button,
                    withLabel && styles['button--padding'],
                    this.props.small && styles['button--small']

                ].join(' ')

            } type={this.props.type} style={{

                color: !this.props.light ? 'white' : `var(--md-${colorDefault})`

            }}>

                <div className={ styles['button__background'] } style={{

                    backgroundColor: this.props.light ? `var(--md-grey-200)` : `var(--md-${colorDefault})`

                }}></div>

                { this.props.label }

            </button>

        )

    }

}

Button.propTypes = {

    disabled: PropTypes.bool,
    loading : PropTypes.bool,

    color   : PropTypes.string,

    light   : PropTypes.bool,
    icon    : PropTypes.string,
    full    : PropTypes.bool,
    small   : PropTypes.bool,

    label   : PropTypes.string,
    type    : PropTypes.string.isRequired

}

Button.defaultProps = {

    type: 'button',

    color: 'blue-500'

}

/* */

export default Button
