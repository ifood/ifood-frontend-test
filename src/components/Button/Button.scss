@import '../../assets/styles/variables';

.button-component {
  display: inline-flex;
  padding: $base-size 1.5rem;
  border: none;
  border-radius: .25rem;
  outline: none;
  font-size: $base-size;
  font-weight: 500;
  text-decoration: none;
  cursor: pointer;
  margin: $base-size 0;
  transition: all .2s ease-in-out;
  &:hover {
    opacity: .8;
  }
  &.default {
    color: white;
    background-color: $primary-color;
  }
}
