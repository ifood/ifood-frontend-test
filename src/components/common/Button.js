import React from 'react'
import dotenv from 'dotenv'

dotenv.config()

const clientId = process.env.REACT_APP_ClientID
const redirectUrl = 'http%3A%2F%2Flocalhost%3A3006%2Fplaylists'

const Button = () => {
    return (
        <a href={`https://accounts.spotify.com/authorize?client_id=${clientId}&redirect_uri=${redirectUrl}&response_type=token&show_dialog=true`}>
            <button className="header__login-btn">Login with Spotify</button>
        </a>
    )
}

export default Button
