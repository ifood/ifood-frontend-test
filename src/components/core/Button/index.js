import React from "react";
import PropTypes from "prop-types";

import { Container, ButtonContent } from "./styles";

const Button = ({
  children,
  handleClick,
  url,
  margin,
  link,
  type,
  ...props
}) => (
  <Container margin={margin}>
    {link ? (
      <ButtonContent
        onClick={handleClick}
        as="a"
        href={url}
        type={type}
        {...props}
      >
        {children}
      </ButtonContent>
    ) : (
      <ButtonContent onClick={handleClick} as="button" type={type} {...props}>
        {children}
      </ButtonContent>
    )}
  </Container>
);

Button.propTypes = {
  children: PropTypes.any.isRequired,
  handleClick: PropTypes.func,
  type: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.oneOf(["primary", "secondary"]),
  ]),
  margin: PropTypes.string,
  url: PropTypes.string,
  link: PropTypes.bool,
};

Button.defaultProps = {
  handleClick: null,
  type: "primary",
  margin: "0",
  url: "",
  link: false,
};

export default Button;
