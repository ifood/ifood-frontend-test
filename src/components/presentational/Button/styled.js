import styled from "styled-components";

import * as C from "constants/styles/colors";
import { SPACINGS } from "constants/styles/spacings";

export const Button = styled.button`
  background-color: ${C.LIGHT};
  color: ${C.PRIMARY_COLOR};
  padding: ${SPACINGS.xs} ${SPACINGS.xl};
  border-radius: 5px;
  outline: none;
  cursor: pointer;
  font-weight: bold;
  font-size: 18px;
  transition: 300ms ease all;

  &:hover {
    background: ${C.PRIMARY_COLOR_LIGHT};
    color: ${C.LIGHT};
  }
`;
