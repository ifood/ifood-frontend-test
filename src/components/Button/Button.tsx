import React from 'react';

import './Button.scss';

import { classesFromObject } from '../../helpers/helpers';

interface IButton {
  className?: string;
  label: string;
  onClick: () => void;
}

const Button = ({
  className: classNameProp = '',
  label,
  onClick,
}: IButton) => {
  const classes = {
    'button-component': true,
  }

  const className = classesFromObject(classes, classNameProp);

  return (
    <button className={className} title={label} onClick={onClick} >
      {label}
    </button>
  );
}

export default Button;
