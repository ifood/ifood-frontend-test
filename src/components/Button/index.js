import React from "react";
import PropTypes from "prop-types";

import * as S from "./styled";

const Button = ({ onClick, label, variation }) => {
  return (
    <S.Button variation={variation} onClick={onClick}>
      {label}
    </S.Button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  variation: PropTypes.oneOf(["primary", "secondary"]).isRequired,
};

export default Button;
