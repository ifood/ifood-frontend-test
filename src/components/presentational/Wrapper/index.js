import React from "react";
import PropTypes from "prop-types";

import * as S from "./styled";

const Wrapper = ({ children }) => {
  return <S.Wrapper>{children}</S.Wrapper>;
};

Wrapper.defaultProps = {
  children: null,
};

Wrapper.propTypes = {
  childrren: PropTypes.node,
};

export default Wrapper;
