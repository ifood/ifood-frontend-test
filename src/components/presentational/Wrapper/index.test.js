import React from "react";
import { render } from "@testing-library/react";

import Button from "components/presentational/Button/index";
import Wrapper from "./index";

test("<Wrapper/> renders correcly", () => {
  const func = () => {
    return true;
  };

  const component = (
    <Wrapper
      childrren={<Button label="Button label" onClick={() => func()} />}
    />
  );
  const { asFragment } = render(component);
  expect(asFragment(component)).toMatchSnapshot();
});
