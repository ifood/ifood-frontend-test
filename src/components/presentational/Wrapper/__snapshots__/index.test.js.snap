// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper/> renders correcly 1`] = `
<DocumentFragment>
  <div
    class="sc-gtssRu gBNuyD"
  />
</DocumentFragment>
`;
